phoenix-core/src/main/java/org/apache/phoenix/util/UpgradeUtil.java

```diff
@@ -1400,6 +1393,43 @@ public static void syncUpdateCacheFreqAllIndexes(PhoenixConnection conn, PTable
         }
     }
 
+    private static void iterateOverChildViewAndSyncCacheFreq(
+            final PhoenixConnection newConn,
+            final PreparedStatement stmt, final TableInfo tableInfo)
+            throws SQLException {
+        String viewName = SchemaUtil.getTableName(tableInfo.getSchemaName(),
+            tableInfo.getTableName());
+        String viewTenantId = Bytes.toString(tableInfo.getTenantId());
```

Comment:
It will be better to resolve the PTable for the view (either using tenanted connection if required or a global connection if a global view) and then call `syncUpdateCacheFreqForIndexesOfTable()` just once. Basically, combine this method and `getViewAndSyncCacheFreqForIndexes()`.